server/metadata/tenant.go

```diff
@@ -374,22 +377,31 @@ func (tenant *Tenant) ReloadUsingOutsideVersion(ctx context.Context, tx transact
 }
 
 func (tenant *Tenant) ReloadUsingTxVersion(ctx context.Context, tx transaction.Tx, id string) error {
-	tenant.Lock()
-	defer tenant.Unlock()
-
 	currentVersion, err := tenant.versionH.Read(ctx, tx)
 	if err != nil {
 		return err
 	}
 
-	if bytes.Equal(currentVersion, tenant.version) {
+	if !tenant.shouldReload(currentVersion) {
```

Comment:
What if the version changes after this check? We don't need to reload?

Reply:
if the version changed after this check then we are good because that means a conflicting transaction has moved forward and this transaction will not see that schema change which is fine.